test/evp_extra_test.c

```diff
@@ -4620,14 +4620,24 @@ static int test_ecx_short_keys(int tst)
 {
     unsigned char ecxkeydata = 1;
     EVP_PKEY *pkey;
+    OSSL_PROVIDER* deflprov;
+    int testresult = 1;
 
 
+    deflprov = OSSL_PROVIDER_load(NULL, "default");
```

Comment:
This should not be necessary. Unless you have a configuration file that explicitly loads only the base and/or legacy provider which would be fairly broken configuration anyway.

None of the other tests explicitly load the default provider.

Reply:
It seems the there's a test case to use a non-default ctx, where the `null` provider is loaded (https://github.com/openssl/openssl/blob/12b1ae38aa4621a85b00f560a3e828f9a5386d04/test/evp_extra_test.c#L4746), although I'm able to pass all tests after remove that line, but I assume some tests are actually "masked" and only looks to be passed.

Reply:
You need to skip the test if testctx is not NULL.

Reply:
That's a good idea, I will update the tests.